examples/notebooks/beam-ml/run_inference_basic.ipynb

```diff
@@ -1362,6 +1308,408 @@
+        "    example = ExampleProcessor().create_example(feature=i)\n",
+        "    writer.write(example.SerializeToString())\n",
+        "\n"
+      ],
+      "metadata": {
+        "id": "t2lQIfeonS0Q"
+      },
+      "execution_count": 8,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Fit The Model\n",
+        "\n",
+        "This example builds a model. However, normal RunInference usage is for pretrained models."
+      ],
+      "metadata": {
+        "id": "qN5RobQ2JN3O"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "# Build the training dataset.\n",
+        "RAW_DATA_TRAIN_SPEC = {\n",
+        "'x': tf.io.FixedLenFeature([], tf.float32),\n",
+        "'y': tf.io.FixedLenFeature([], tf.float32)\n",
+        "}\n",
+        "\n",
+        "RAW_DATA_PREDICT_SPEC = {\n",
+        "'x': tf.io.FixedLenFeature([], tf.float32),\n",
+        "}\n",
+        "\n",
+        "dataset = tf.data.TFRecordDataset(example_five_times_table)\n",
+        "dataset = dataset.map(lambda e : tf.io.parse_example(e, RAW_DATA_TRAIN_SPEC))\n",
+        "dataset = dataset.map(lambda t : (t['x'], t['y']))\n",
+        "dataset = dataset.batch(100)\n",
+        "dataset = dataset.repeat()\n",
+        "\n",
+        "# Fit the model\n",
+        "model.fit(dataset, epochs=500, steps_per_epoch=1, verbose=0)"
+      ],
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "03OVIGDtn7WI",
+        "outputId": "b1eef9a4-b5db-42e3-8f8a-fd3da71f8c38"
+      },
+      "execution_count": 13,
+      "outputs": [
+        {
+          "output_type": "execute_result",
+          "data": {
+            "text/plain": [
+              "<keras.callbacks.History at 0x7f5a3f8eebd0>"
+            ]
+          },
+          "metadata": {},
+          "execution_count": 13
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Save the Model"
+      ],
+      "metadata": {
+        "id": "KtrqJ6i2JmAS"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "@tf.function(input_signature=[tf.TensorSpec(shape=[None], dtype=tf.string , name='examples')])\n",
+        "def serve_tf_examples_fn(serialized_tf_examples):\n",
+        "  \"\"\"Returns the output to be used in the serving signature.\"\"\"\n",
+        "  features = tf.io.parse_example(serialized_tf_examples, RAW_DATA_PREDICT_SPEC)\n",
+        "  return model(features, training=False)\n",
+        "\n",
+        "signature = {'serving_default': serve_tf_examples_fn}\n",
+        "\n",
+        "tf.keras.models.save_model(model, save_model_dir_multiply, signatures=signature)"
+      ],
+      "metadata": {
+        "id": "pscBzhsakGnr"
+      },
+      "execution_count": 17,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "## Run the Pipeline\n",
```

Comment:
I'd love to see another section or 2 on common patterns (this could mirror PyTorch a little bit, though it doesn't have to cover the same ground - e.g. an A/B example could be neat, though it doesn't need to be that specifically.)

Reply:
I think this is a good idea. With tensorflow there are some other interesting patterns that are available to illustrate. Specifically:
 - How to use tfx-bsl with Vertex AI
 - An example with a classification.
 - A mixed, regression/classification example.

I want to do that in a separate PR if possible since it will be a little more involved and I think this is a good start.